tsconfig.json

```diff
@@ -0,0 +1,41 @@
+{
+	"compilerOptions": {
+		"baseUrl": ".",
+		"module": "commonjs",
+		"moduleResolution": "node",
+		"lib": [ "dom", "esnext" ],
+		"allowJs": true,
+		"checkJs": true,
+		"noEmit": true,
+		"traceResolution": true,
+		"strictNullChecks": true,
+		"noUnusedLocals": true,
+		"noUnusedParameters": true,
+		"noImplicitReturns": true,
+		"noFallthroughCasesInSwitch": true,
+		"jsx": "preserve",
+		"allowSyntheticDefaultImports": true,
+		"paths": {
+			"@woocommerce/atomic-components/*": [
+				"assets/js/base/atomic/components/*"
+			],
+			"@woocommerce/base-components/*": [ "assets/js/base/components/*" ],
+			"@woocommerce/base-context/*": [ "assets/js/base/context/*" ],
+			"@woocommerce/base-hocs/*": [ "assets/js/base/hocs/*" ],
+			"@woocommerce/base-hooks/*": [ "assets/js/base/hooks" ],
+			"@woocommerce/base-utils": [ "assets/js/base/utils" ],
+			"@woocommerce/block-components/*": [ "assets/js/components/*" ],
+			"@woocommerce/block-hocs/*": [ "assets/js/hocs/*" ],
+			"@woocommerce/blocks-registry/*": [ "assets/js/blocks-registry/*" ],
+			"@woocommerce/settings": [ "assets/js/settings/shared" ],
+			"@woocommerce/block-settings": [ "assets/js/settings/blocks" ],
+			"@woocommerce/icons": [ "assets/js/icons" ],
+			"@woocommerce/resource-previews": [ "assets/js/previews" ],
+			"@woocommerce/e2e-tests": [
```

Comment:
Will need to remember to keep this up to date. Is there a way we can share this with webpack or other config?

Reply:
> Is there a way we can share this with webpack or other config?

Currently not easily because it's not a 1:1 match with other configs (and this is strict json vs js). There may still be some adjusting needed with accounting for legacy build source paths (which we aren't using yet). 

All of this can be done as needed when discovered though.